Update readme (#48)
kubernetes-sigs/kwok: README.md

Kwokctl is a tool for creating fake clusters of nodes that are simulated by Kwok.

## Examples

These examples are tested to work with the latest released version of Kwok/Kwokctl.

### Basic Usage

- [Kwok Manage Nodes and Pods](./docs/examples/kwok-manage-nodes-and-pods.md) - Kwok's basic management of Node and Pod operations
- Kwok - maintain Nodes heartbeat and Pods status.
  - [Kwok in Cluster](./docs/examples/kwok-in-cluster.md) - Installing Kwok in a cluster
  - [Kwok in Local](./docs/examples/kwok-in-local.md) - Run Kwok in the local for a cluster
- Kwokctl - cluster creation, etcd snapshot, etc.
  - [Kwokctl Manage Clusters](./docs/examples/kwokctl-manage-cluster.md) - Create/Delete a cluster in local where all nodes are managed by Kwok
  - [Kwokctl Snapshot Clsuter](./docs/examples/kwokctl-snapshot.md) - Save/Restore the Etcd data of a cluster created by Kwokctl

``` console
$ time kwokctl create cluster
Creating cluster "kwok-kwok"
Starting cluster "kwok-kwok"
Cluster "kwok-kwok" is ready
You can now use your cluster with:

    kubectl config use-context kwok-kwok

Thanks for using kwok!

real    0m2.599s
user    0m0.606s
sys     0m0.254s
```

> ### Actual Usage

Author: Not an advanced use case
Since Kwok is a new project, it would help to promote kwok if there is a friendly link with other projects (like https://github.com/kubernetes-sigs/external-dns#the-latest-release)

For advanced, we can later summarize the scenarios that users use more and then add

> ### Actual Usage

Reviewer: so like "User Stories"?

> ### Actual Usage

Author: Yes, "User Stories" and "Friendly link"

If you are using Kwok/Kubectl as a testing or CI in your project and would like to share your experience with others, then please add your example below

<!--
Add your examples like

- [Example name](./docs/examples/example-name/example-name.md) - Example description
-->

- [TBD](#) - Example description

## Kwokctl Runtime and OS Support

|         | binary | docker | kind | nerdctl | podman |
| ------- | ------ | ------ | ---- | ------- | ------ |
| Linux   | 🟢      | 🟢      | 🟢    | 🔵       | ⚫      |
| Darwin  | 🟠      | 🟢      | 🟢    | 🔴       | 🔴      |
| Windows | 🟠/🔵    | 🔵      | 🔵    | 🔴       | 🔴      |

- 🟢 Support
- 🔴 Not support
- 🟠 Need to use your own build of the Kubernetes binary
- 🔵 Expected support but not fully tested
- ⚫ TODO

### Kwokctl Default Runtime

> - Linux: binary

Author: https://www.downloadkubernetes.com/

k8s only provides binaries for control plane components under Linux. Windows and Darwin need to be compiled themselves.

> - Linux: binary

Reviewer: My point is that images is the de-factor standard of running control plane components, and gcr.k8s.io has prebuilt vanilla images, so why not just use `docker` as the default binary for all platforms.

> - Linux: binary

Author: That's fine. Let me update that. #51

# KWOK (Kube-WithOut-Kubelet)

> The repository is a developer's toolkit that supports the fast startup of a cluster of K-ish nodes.

Reviewer: ```suggestion
The repository is a toolkit that enables setting up a cluster of thousands of Nodes in seconds.
Under the scene, all Nodes are simulated to behave like real ones, so the overall approach employes
a pretty low resource footprint that you can easily play around on your laptop.
```

> Simulate thousands of fake kubelets, on a laptop with minimum resource footprint.

Reviewer: remove this line.

> The repository has 2 tools:

Reviewer: ```suggestion
So far we provide two tools:
```

> - Kwok is used to simulate thousands of fake nodes.
> - Kwokctl is used to create clusters of fake nodes simulated by Kwok.

Reviewer: ```suggestion
- **Kwok:** core of this repo. It simulates thousands of fake Nodes.
- **Kwokctl:** A CLI to facilitate creating and managing clusters simulated by Kwok.
```
